gcc/rust/typecheck/rust-hir-type-check-expr.h

```diff
@@ -387,43 +387,6 @@ class TypeCheckExpr : public TypeCheckBase
     auto result = lhs->coerce (rhs);
     if (result->get_kind () == TyTy::TypeKind::ERROR)
       return;
```

Comment:
Hm, if this is the new end of the function are these lines doing anything anymore?

Reply:
Oh yeah good point will fix that.

The type system has loads of side effects on the type check context table, because of the nature of inference variables which upon any type inference will walk the references and update them accordingly.